src/Features/CSharp/Portable/StackTraceExplorer/CSharpStackTraceExplorerService.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.StackTraceExplorer;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.StackTraceExplorer
+{
+    [Shared]
+    [ExportLanguageService(typeof(IStackTraceExplorerService), language: LanguageNames.CSharp)]
+    internal class CSharpStackTraceExplorerService : IStackTraceExplorerService
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpStackTraceExplorerService()
+        {
+        }
+
+        public string GetClassMetadataName(string className)
+        {
+            var typeNameSyntax = SyntaxFactory.ParseTypeName(className);
```

Comment:
i would def doc the expected forms of classnames we're getting.  are we genuinely expecting them to be 100% legal c# type names?

Reply:
Added documentation here that these tie directly to the TypeSpan/MethodSpan in ParsedStackFrame